tests/test_black.py

```diff
@@ -595,6 +595,15 @@ def test_write_cache_write_fail(self) -> None:
             mock.side_effect = OSError
             black.write_cache({}, [])
 
+    def test_check_diff_use_together(self) -> None:
```

Comment:
Instead of creating a temp file (which is slow), just run -the command on two existing files. I suggest`tests/string_quotes.py` and `tests/composition.py`. The former should return exit code 1, the latter should return exit code 0. The former should produce a diff, the latter should not produce a diff.